### PR TITLE
Using protoc-gen-doc from the flyteorg and regenerated docs for new modules

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR

- Forked pseudomuto/protoc-gen-doc in flyteorg/protoc-gen-doc
- Merged the RST supported pull request in this new forked repo
- Released v1.4.2  of protoc-gen-doc from new repo
- Updated flyteidl to use and install this version through boilerplate
- Tested using make generate and saw that the protoc-gen-doc plugin is getting installed 
- Regenerated the docs which are now updated core.rst and plugins.rst for new changes (Need to automate this to happen during code checkin ) - K8sObjectMetadata, SideCar changes

(Note : Earlier code change in boilerplate would say its installed but wont reflect the binary generation. This was fixed by providing the path to main github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc instead of just github.com/pseudomuto/protoc-gen-doc. Thanks @evalsocket for helping find this issue)

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Review by katrogan — DISMISSED** [review]
awesome, thank you for fixing! just to check `make generate` should now fully succeed in generating these docs in the future?

**Comment by pmahindrakar-oss** [comment]
@katrogan make generate already generates these docs but those are not caught by the githubworkflows since they individually generate the protos 

```
# GO Protos
      - name: Proto-Service-Go
        run: python3 /usr/local/bin/entrypoint.py -i ./protos -d protos/flyteidl/service --with_gateway -l go --go_source_relative
      - name: Proto-Admin-Go
        run: python3 /usr/local/bin/entrypoint.py -i ./protos -d protos/flyteidl/admin --with_gateway -l go --go_source_relative --validate_out
      - name: Proto-Core-Go
        run: python3 /usr/local/bin/entrypoint.py -i ./protos -d protos/flyteidl/core --with_gateway -l go --go_source_relative --validate_out
      - name: Proto-Event-Go
        run: python3 /usr/local/bin/entrypoint.py -i ./protos -d protos/flyteidl/event --with_gateway -l go --go_source_relative --validate_out
      - name: Proto-Plugins-Go
        run: python3 /usr/local/bin/entrypoint.py -i ./protos -d protos/flyteidl/plugins -l go --go_source_relative --validate_out
      - name: Proto-Datacatalog-Go
        run: python3 /usr/local/bin/entrypoint.py -i ./protos -d protos/flyteidl/datacatalog -l go --go_source_relative --validate_out
```
More details in this link to github workflow
https://github.com/flyteorg/flyteidl/blob/bcfcc3df92168376f083294aef639c8f42450ee2/.github/workflows/verification.yml#L49

We would need to change the github workflows to do make generate. I will test it out separately and send out a PR.

**Review by evalsocket — APPROVED** [review]

File: boilerplate/lyft/golang_test_targets/download_tooling.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ tools=(
   "github.com/lyft/flytestdlib/cli/pflags"
   "github.com/golangci/golangci-lint/cmd/golangci-lint"
   "github.com/alvaroloes/enumer"
-  "github.com/pseudomuto/protoc-gen-doc"
+  "github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc"
```
Comment (kumare3) [inline boilerplate/lyft/golang_test_targets/download_tooling.sh:23]: you have to add this to boilerplate. When boilerplate update this will be overriden

Comment (kumare3) [reply boilerplate/lyft/golang_test_targets/download_tooling.sh:23]: you will have to add it here - https://github.com/flyteorg/boilerplate/blob/master/boilerplate/flyte/golang_support_tools/tools.go

cc @evalsocket

**Review by kumare3 — CHANGES_REQUESTED** [review]

File: boilerplate/lyft/golang_test_targets/download_tooling.sh (hunk 1 above)
Comment (pmahindrakar-oss) [reply boilerplate/lyft/golang_test_targets/download_tooling.sh:23]: Done here https://github.com/flyteorg/boilerplate/pull/27

Comment (evalsocket) [reply boilerplate/lyft/golang_test_targets/download_tooling.sh:23]: @pmahindrakar-oss @kumare3 I also have one pending pr related to boilerplate https://github.com/flyteorg/flyteidl/pull/162 please merge it before merging it

Comment (pmahindrakar-oss) [reply boilerplate/lyft/golang_test_targets/download_tooling.sh]: Awaiting the PR https://github.com/flyteorg/boilerplate/pull/27 to be merged and updated in boilerplate for flyteidl automatically

**Review by kumare3 — APPROVED** [review]

**Review by katrogan — APPROVED** [review]
